components/gcp/dataflow/launch_template/sample.ipynb

```diff
@@ -4,89 +4,126 @@
+    "\n",
+    "## Cautions and requirements\n",
+    "To use the components, the following requirements must be met:\n",
+    "* Dataflow API is enabled.\n",
+    "* The component is running under a secret [Kubeflow user service account](https://www.kubeflow.org/docs/started/getting-started-gke/#gcp-service-accounts) in a KFP cluster. For example:\n",
+    "```\n",
+    "component_op(...).apply(gcp.use_gcp_secret('user-gcp-sa'))\n",
+    "```\n",
+    "The Kubeflow user service account is a member of `roles/dataflow.developer` role of the project.\n",
+    "The Kubeflow user service account is a member of `roles/storage.objectViewer` role of the Cloud Storage Object `gcs_path`.\n",
+    "The Kubeflow user service account is a member of `roles/storage.objectCreator` role of the Cloud Storage Object `staging_dir`.\n",
+    "\n",
+    "## Detailed description\n",
+    "The input `gcs_path` must contain a valid Dataflow template. The template can be created by following the guide [Creating Templates](https://cloud.google.com/dataflow/docs/guides/templates/creating-templates). Or, you can use [Google-provided templates](https://cloud.google.com/dataflow/docs/guides/templates/provided-templates).\n",
+    "\n",
+    "Here are the steps to use the component in a pipeline:\n",
+    "1. Install KFP SDK\n",
+    "Install the SDK (Uncomment the code if the SDK is not installed before)"
    ]
   },
   {
-   "cell_type": "markdown",
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {},
+   "outputs": [],
    "source": [
-    "## Sample\n",
+    "%%capture\n",
     "\n",
-    "Note: the sample code below works in both IPython notebook or python code directly."
+    "KFP_PACKAGE = 'https://storage.googleapis.com/ml-pipeline/release/0.1.14/kfp.tar.gz'\n",
+    "!pip3 install $KFP_PACKAGE --upgrade"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Set sample parameters"
+    "2. Load the component by DSL"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "tags": [
-     "parameters"
-    ]
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "# Required Parameters\n",
-    "PROJECT_ID = '<Please put your project ID here>'\n",
-    "GCS_WORKING_DIR = 'gs://<Please put your GCS path here>' # No ending slash\n",
+    "import kfp.components as comp\n",
     "\n",
-    "# Optional Parameters\n",
-    "EXPERIMENT_NAME = 'Dataflow - Launch Template'\n",
-    "COMPONENT_SPEC_URI = 'https://raw.githubusercontent.com/kubeflow/pipelines/d2f5cc92a46012b9927209e2aaccab70961582dc/components/gcp/dataflow/launch_template/component.yaml'"
+    "COMPONENT_SPEC_URI = 'https://raw.githubusercontent.com/kubeflow/pipelines/d2f5cc92a46012b9927209e2aaccab70961582dc/components/gcp/dataflow/launch_template/component.yaml'\n",
+    "dataflow_template_op = comp.load_component_from_url(COMPONENT_SPEC_URI)\n",
```

Comment:
Do we really need two lines to load component? Having a single line will make it much easier to copy/paste (otherwise, the variable name is duplicate).

Reply:
Done